### DISCUSSION TITLE
set Node ID available only in apps with data persistence

### DISCUSSION THREAD

**Discussion by erickythierry** [caprover/caprover#2143] [Q&A] [answered]
Greetings. I would like to know if there is any technical limitation that prevents me from changing the node ID of apps that do not have the data persistence flag?

I recently migrated to cluster mode and tried to move some simple apps that do not have the persistence flag, but the option to edit the node ID does not appear in them.

However, in apps that do have this persistence flag, it is possible to change the node ID.

Is this a bug?

Another question: my main server is ARM, and the worker is x86. So any app I deploy ends up using ARM as a base, making it impossible to run the image on the x86 worker.
If I edit the Push Registry to "disabled push", would the worker produce its own image? Or would there be a way to get around this difference in architectures?

**Comment by githubsaturn** [comment, marked as answer]
Not a bug but a feature. For a stateless application, Docker Swarm automatically deploys the app to the optimum node to ensure good workload balance etc. Ideally, you want to do the same for persistent apps too, but unfortunately since the data is on disk, they have to be locked down to a specific node. So it's more of a limitation for the persistent apps, rather than a limitation for the stateless apps. 

If you really really want to lock down a stateless app to a node, you can do so by running:
```
docker service update --constraint-add node.id==<NodeId> <srv-captain--yourservice>
